stdlib/bytes.gr

```diff
@@ -0,0 +1,505 @@
+ * @param v: Int64 - The value to set
+ * @returns Void
+ */
+@disableGC
+export let setInt64 = (b: Bytes, i: Number, v: Int64) => {
+  let byteSize = 8n;
+  let (ptr, offset) = getPointerAndOffset(b, byteSize, i);
+  let value = int64ToWasmI64(v);
+  WasmI64.store(ptr, value, offset);
+}
+
+/**
+ * Gets a signed 64-bit integer starting at the given byte index.
+ * @param b: Bytes - The Bytes buffer
+ * @param i: Number - The byte index
+ * @returns Int64
+ */
+@disableGC
+export let getInt64 = (b: Bytes, i: Number) => {
+  let byteSize = 8n;
+  let (ptr, offset) = getPointerAndOffset(b, byteSize, i);
+  let n = WasmI64.load(ptr, offset);
+  wasmI64ToInt64(n)
+}
+
+/**
+ * Sets a 64-bit float starting at the given byte index.
+ * @param b: Bytes - The Bytes buffer
+ * @param i: Number - The byte index
+ * @param v: Float64 - The value to set
+ * @returns Void
+ */
+@disableGC
+export let setFloat64 = (b: Bytes, i: Number, v: Float64) => {
+  let byteSize = 8n;
+  let (ptr, offset) = getPointerAndOffset(b, byteSize, i);
+  let value = float64ToWasmF64(v);
+  WasmF64.store(ptr, value, offset);
+}
+
+/**
+ * Gets a 64-bit float starting at the given byte index.
+ * @param b: Bytes - The Bytes buffer
+ * @param i: Number - The byte index
+ * @returns Float64
+ */
+@disableGC
+export let getFloat64 = (b: Bytes, i: Number) => {
+  let byteSize = 8n;
+  let (ptr, offset) = getPointerAndOffset(b, byteSize, i);
+  let n = WasmF64.load(ptr, offset);
+  wasmF64ToFloat64(n)
+}
+
+/**
+ * Return a new Bytes buffer that contains the same bytes as the argument.
+ * @param b: Bytes - The Bytes buffer to copy
+ * @returns Bytes
+ */
+@disableGC
+export let copy = (b: Bytes) => {
+  let (<) = WasmI32.ltS;
+  let (+) = WasmI32.add;
+  let (src, size) = getPointerAndSize(b);
+  let dst = allocateBytes(size);
+  for (let mut i = 0n; i < size; i += 4n) {
+    let n = WasmI32.load(src, 8n + i);
+    WasmI32.store(dst, n, 8n + i);
+  }
+  WasmI32.toGrain(dst): Bytes
+}
+
+/**
+ * Returns a new Bytes buffer containing a subset of the original Bytes buffer.
+ * @param b: Bytes - The Bytes buffer get a subset of bytes from
+ * @param i: Number - The start position to copy from
+ * @param len: Number - The number of bytes to copy
+ * @returns Bytes
+ */
+@disableGC
+export let sub = (b: Bytes, i: Number, len: Number) => {
+  let (>) = WasmI32.gtS;
+  let (%) = WasmI32.remS;
+  let (==) = WasmI32.eq;
+  let (<) = WasmI32.ltS;
+  let (+) = WasmI32.add;
+  let (src, size) = getPointerAndSize(b);
+  let i = numberToWasmI32(i);
+  let len = numberToWasmI32(len);
+  if ((i + len) > size) {
+    throw Exception.InvalidArgument("The given index and length do not specify a valid range of bytes");
+  }
+  let dst = allocateBytes(len);
+  let offset = i;
+  if (i % 4n == 0n) {
+    // If possible, copy 4 bytes at a time
+    for (let mut i = 0n; i < len; i += 4n) {
+      let n = WasmI32.load(src, 8n + i + offset);
+      WasmI32.store(dst, n, 8n + i);
```

Comment:
I got this far down and didn't catch it until now, but it's a tad more efficient to write loads/stores with a static offset:
```suggestion
      WasmI32.store(dst + i, n, 8n);
```